### ISSUE TITLE
Not working in Gatsby production mode

### ISSUE THREAD

**Issue by ryan-ds** [silvenon/gatsby-plugin-linaria#17]
## Environment

node: `8.12.0`
gatsby: `2.12.0`
gatsby-plugin-linaria: `1.2.0`
linaria: `1.3.1`
react: `16.8.6`

## Description

- I have created a new starter project with Gatsby via the following command:
`npx gatsby new my-default-starter https://github.com/gatsbyjs/gatsby-starter-default`

- Then added the required packages via:
`yarn add gatsby-plugin-linaria linaria`

- After that, I added the `gatsby-plugin-linaria` plugin at the end of plugins in the `gatsby-config.js` file.

- Lastly I edited the `components/header.js` file to reflect the following:
```js
import { Link } from 'gatsby'
import { styled } from 'linaria/react'
import PropTypes from 'prop-types'
import React from 'react'

const MainHeader = styled.header`
  background: rebeccapurple;
  margin-bottom: 1.45rem;
`

const Header = ({ siteTitle }) => (
  <MainHeader>
    <div
      style={{
        margin: `0 auto`,
        maxWidth: 960,
        padding: `1.45rem 1.0875rem`,
      }}
    >
      <h1 style={{ margin: 0 }}>
        <Link
          to="/"
          style={{
            color: `white`,
            textDecoration: `none`,
          }}
        >
          {siteTitle}
        </Link>
      </h1>
    </div>
  </MainHeader>
)

Header.propTypes = {
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
```

---

After following these steps, I ran the project in development via `npm start` and it works flawlessly.

But running in production via `npm run build && npm run serve`, the build is successful but the styles do not get applied. The class is applied but the styles are not reflected anywhere.

![image](https://user-images.githubusercontent.com/38418557/60515763-f7d17580-9cf9-11e9-9e9d-7329b2017ff9.png)

I also tried deleting the `public`, `.cache` and `linaria-cache` folders but it didn't make any difference.

**Comment by ryands17** [comment]
@silvenon I guess the issue is in the `gatsby-ssr.js` file where for some reason this line is not being executed
```js
if (type === 'style' && LINARIA_STYLESHEET_RULE.test(props['data-href'])) {
  // code here is not executed
}
```

The regex doesn't seem to match meaning it's not getting the required linaria.css file in the `getHeadComponents` method.

**Comment by silvenon** [comment]
Does [v1.2.1](https://github.com/silvenon/gatsby-plugin-linaria/releases/tag/v1.2.1) fix the issue?

**Comment by ryands17** [comment]
@silvenon Nope, version `1.2.1` has the same issue. In the `getHeadComponents` method, I tried logging the `props` argument and it doesn't have contain any css file matching the `LINARIA_STYLESHEET_RULE` regex.

**Comment by cometkim** [comment]
The gatsby-ssr impl only adjusts the priority for the `<link>` matched to the regex, so codes inside filter not being executed means there is no effect.

There may be another (something like loader changed inside of gatsby?) cause.

**Comment by cometkim** [comment]
Options for enable/disable critical css would be nice BTW.

**Comment by ryan-ds** [comment]
@cometkim @silvenon yes if I omit the `gatsby-ssr.js` file, it still doesn't have any effect, so I guess the issue is regarding the loaders changed in production mode

**Comment by ryan-ds** [comment]
@cometkim @silvenon if possible could you pls suggest a version of this plugin as well as the `gatsby` and `linaria` versions working together so that I can test the application which I am refactoring currently

**Comment by cometkim** [comment]
@ryan-ds @silvenon gotcha, 
![image](https://user-images.githubusercontent.com/9696352/60605564-ba054780-9df4-11e9-86aa-c9f46e71d608.png)

linaria doesn't load properly since gatsby@2.12 (latest of 2.11 is ok)

**Comment by cometkim** [comment]
https://github.com/gatsbyjs/gatsby/commit/268ed273a5e8be51a9bd38e2fd5d6cc55b1c127d found this.

**Comment by cometkim** [comment]
Working on this. it would be resolved with a few cleanups.

**Comment by cometkim** [comment]
done. :sunglasses:

**Comment by ryands17** [comment]
@cometkim tried by cloning your branch and running the build but it fails with this error
![image](https://user-images.githubusercontent.com/19697099/60621834-61d24380-9dfc-11e9-8bfd-f6d1b0534aee.png)

**Comment by cometkim** [comment]
@ryands17 I think it just have been fixed in last commit. Could you try it again?

**Comment by ryan-ds** [comment]
@cometkim It works flawlessly. Thanks a lot :)

**Comment by ryands17** [comment]
It works flawlessly. Thanks a lot :)

**Comment by silvenon** [comment]
Published in [v1.2.2](https://github.com/silvenon/gatsby-plugin-linaria/releases/tag/v1.2.2) 🚀
